core/rust/qdbr/src/parquet/error.rs

```diff
@@ -114,6 +124,22 @@ impl ParquetError {
     pub fn display_with_backtrace(&self) -> ParquetErrorWithBacktraceDisplay {
         ParquetErrorWithBacktraceDisplay(self)
     }
+
+    fn build_out_of_memory() -> ParquetError {
+        let last_err = take_last_alloc_error();
+        let no_last_err = last_err.is_none();
```

Comment:
Could you explain the scenario where there would be no last allocation error?

Reply:
Mostly theoretical, but it could happen that the thread local is cleared before it's accessed.

I'd rather handle such theoretical case, rather than `panic!`.